checkstyle.xml

```diff
@@ -67,7 +67,10 @@
         <module name="LineLength">
             <property name="max" value="120"/>
         </module>
-        <module name="MethodLength"/>
+	<module name="MethodLength">
```

Comment:
I think we should split it. The easiest option I can see is to extract 3 different methods from `if` conditions.